### DISCUSSION TITLE
关于项目本地的插件是否能确保插件加载和跨插件访问的问题

### DISCUSSION THREAD

**Discussion by Jerry-FaGe** [nonebot/discussions#220] [问答] [answered]
举个栗子，我项目的 `src/plugins` 目录中有 `a_plugin` 和 `b_plugin` 两个插件并且在 `pyproject.toml` 中声明了 `plugin_dirs = ["src/plugins"]`。现在我想让 `a_plugin` 在启动后访问 `b_plugin` 中的方法。于是我按照文档给 `a_plugin` 加入了如下代码

```python
from nonebot import require

require('b_plugin')
from b_plugin import some_fun
```

但是当我重启程序的时候她报了 `No module named 'b_plugin'` 错误，报错在第四行，也就是说第三行的代码似乎并没有确保 `b_plugin` 先加载完成了。另外经过测试 pypi 上安装的插件确实可以通过这个方法正常互相访问。请问本地插件出现这种问题是我操作不当还是 nonebot 的特性呢，我该如何解决这个问题？

**Comment by yanyongyu** [comment, marked as answer]
直接`from src.plugins.a_plugin import xxxx` 就行，不需要require，require通常是第三方库用的

**Reply by Jerry-FaGe** [reply]
感谢！确实是这样！不过还是建议在文档中说明一下这种情况，现在的文档看起来还挺有误导性的（
